WordPress/gutenberg
# GradientPicker: refactor to TypeScript

## What?
Refactor `GradientPicker` component to TypeScript

Part of https://github.com/WordPress/gutenberg/issues/35744

## Why?

The refactor to TypeScript has many benefits (auto-generated docs, static linting and error prevention, better IDE experience). See https://github.com/WordPress/gutenberg/issues/35744 for more details

## How?
Followed the steps in the [TypeScript migration guide](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md?rgh-link-date=2022-11-29T14%3A45%3A34Z#refactoring-a-component-to-typescript)

## Testing Instructions
- Ensure all unit and e2e tests still pass
- Test the component in the editor (e.g. on the background of a Cover block) and Storybook to confirm there are no unexpected changes.
